test/bin/opam-provided.t/run.t

```diff
@@ -0,0 +1,125 @@
+  "base-bigarray" {= "base"}
+  "base-threads" {= "base"}
+  "base-unix" {= "base"}
+  "dune" {= "2.9.1"}
+  "ocaml" {= "4.13.1"}
+  "ocaml-base-compiler" {= "4.13.1"}
+  "ocaml-config" {= "2"}
+  "ocaml-options-vanilla" {= "1"}
+
+Let's now check with the same Opam file but this one adds `opam`-provided.
+Asking for the value will return that "b" will be provided by Opam:
+
+  $ opam show --no-lint --raw -fx-opam-monorepo-opam-provided ./opam-provided.opam
+  "b"
+  $ opam-monorepo lock opam-provided > /dev/null
+
+We should see that this package is not marked as `vendor` so if we run
+`opam` on it, it will install the package "b".
+
+  $ opam show --no-lint --raw -fdepends ./opam-provided.opam.locked
+  "b" {= "1"}
+  "base-bigarray" {= "base"}
+  "base-threads" {= "base"}
+  "base-unix" {= "base"}
+  "dune" {= "2.9.1"}
+  "ocaml" {= "4.13.1"}
+  "ocaml-base-compiler" {= "4.13.1"}
+  "ocaml-config" {= "2"}
+  "ocaml-options-vanilla" {= "1"}
+
+What happens in the case that a package would be ok to vendor but the
+transitive dependency is `opam`-provided? In this case we have a package
+`transitive` that depends on packages that will
+depend transitively on an `opam`-provided package.
+
+  $ opam show --no-lint --raw -fdepends ./transitive.opam
+  "dune" "depends-on-b"
+  $ opam show --no-lint --raw -fx-opam-monorepo-opam-provided ./transitive.opam
+  "b"
+
+Locking it should work as usual
+
+  $ opam-monorepo lock transitive > /dev/null
+  $ opam show --no-lint --raw -fdepends ./transitive.opam.locked
+  "b" {= "1"}
+  "base-bigarray" {= "base"}
+  "base-threads" {= "base"}
+  "base-unix" {= "base"}
+  "depends-on-b" {= "1" & vendor}
+  "dune" {= "2.9.1"}
+  "ocaml" {= "4.13.1"}
+  "ocaml-base-compiler" {= "4.13.1"}
+  "ocaml-config" {= "2"}
+  "ocaml-options-vanilla" {= "1"}
+
+`depends-on-b` is vendored (since it can) but "b" is `opam`-provided. Neat.
+
+Now for the reverse case, `depends-on-b` is `opam`-provided.
+
+  $ opam show --no-lint --raw -fdepends ./reverse-transitive.opam
+  "dune" "depends-on-b"
+  $ opam show --no-lint --raw -fx-opam-monorepo-opam-provided ./reverse-transitive.opam
+  "depends-on-b"
+
+Since it is, we need to make its dependency, "b", also `opam`-provided.
+
+  $ opam-monorepo lock reverse-transitive > /dev/null
+  $ opam show --no-lint --raw -fdepends ./reverse-transitive.opam.locked
+  "b" {= "1"}
+  "base-bigarray" {= "base"}
+  "base-threads" {= "base"}
+  "base-unix" {= "base"}
+  "depends-on-b" {= "1"}
+  "dune" {= "2.9.1"}
+  "ocaml" {= "4.13.1"}
+  "ocaml-base-compiler" {= "4.13.1"}
+  "ocaml-config" {= "2"}
+  "ocaml-options-vanilla" {= "1"}
+
+Since we add a new custom stanza to the Opam file, let's make sure we emit
+warnings when things are specified the wrong way, e.g., if the package to
+be ignored is not a string.
+
+  $ opam show --no-lint --raw -fx-opam-monorepo-opam-provided ./warning.opam
+  42
+
+  $ opam-monorepo lock warning > /dev/null
+  opam-monorepo: String or List required
```

Comment:
Shouldn't the location in the file be printed here?

Reply:
That's a a good point. I looked how the other code does this and adjusted the code to match. Quite neat functionality.